integration/gitopscreateprofile_test.go

```diff
@@ -0,0 +1,90 @@
+// +build integration
+
+package integration_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gexec"
+	"github.com/spf13/afero"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+var _ = Describe("(Integration) generate profile", func() {
+
+	testDirectory := "test_profile"
+
+	BeforeSuite(func() {
+		kubeconfigTemp = false
+		if kubeconfigPath == "" {
+			wd, _ := os.Getwd()
+			f, _ := ioutil.TempFile(wd, "kubeconfig-")
+			kubeconfigPath = f.Name()
+			kubeconfigTemp = true
+		}
+	})
+
+	AfterSuite(func() {
+		gexec.KillAndWait()
+		os.RemoveAll(testDirectory)
+	})
+
+	Describe("when generating a profile", func() {
+		It("should write the processed repo files in the supplied directory", func() {
+
+			clusterName = "amazing-testing-gopher"
+
+			eksctlSuccess("generate", "profile",
+				"--verbose", "4",
+				"--name", clusterName,
+				"--region", region,
+				"--git-url", "git@github.com:eksctl-bot/eksctl-profile-integration-tests.git",
+				"--profile-path", testDirectory,
+			)
+
+			fs := afero.Afero{
+				Fs: afero.NewOsFs(),
+			}
+
+			contents, err := fs.ReadFile(filepath.Join(testDirectory, "workloads/namespace.yaml"))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(string(contents)).To(Equal(
```

Comment:
Actually, you want to use `MatchYAML` here.

```
Expect(contents).To(MatchYAML(expected))
```
Where `contents` and `expected` can be `string`, or `[]byte`, as long as both are the same type.

Reply:
wooooo Nice! thanks, I will change it